### ISSUE TITLE
uncurry should not be defined by pattern matching

### ISSUE THREAD

**Issue by jcommelin** [leanprover-community/lean#162]
https://github.com/leanprover-community/lean/blob/ec1613aef1eee72e601f192b16740629c6d49690/library/init/function.lean#L133

The current definition is bad for definitional reductions. It should be defined using projections.

**Comment by cipher1024** [comment]
I agree

**Comment by gebner** [comment]
:+1:

**Comment by urkud** [comment]
https://github.com/leanprover-community/lean/pull/161/files#diff-42c23da308a4d5900f9f3244953701da

**Comment by urkud** [comment]
It's a part of #160, pending PR #161

**Comment by cipher1024** [comment]
I think we should merge before closing issues.

**Comment by urkud** [comment]
I closed because it is a duplicate of #160, not because we have #161 pending.
